### DISCUSSION TITLE
Could I get issue title in Github actions by environment variables?

### DISCUSSION THREAD

**Discussion by KouWakai** [community/community#26715] [Actions]
<p>Hi I’m trying to get issue title to use it to create issue clone on an another service.</p>
<p>If it’s possible I’d like to get issue title in yml like getting branch name ?</p>

**Comment by ViacheslavKudinov** [comment]
<p>Hi <a class="mention" href="/kouwakai">@KouWakai</a><br>
unfortunately, there is no that variable.<br>
You can check what are in place <a href="https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables" class="inline-onebox" rel="noopener nofollow ugc">Environment variables - GitHub Docs</a>.<br>
Also you can try to check GH marketplace <a href="https://github.com/marketplace?type=actions&amp;query=issue" class="inline-onebox" rel="noopener nofollow ugc">GitHub Marketplace · Actions to improve your workflow · GitHub</a> maybe some actions can help you to achieve your goals.</p>

**Comment by jsoref** [comment]
<p>Depending on the event, it’s possible that what you want is available in the event objects which can be mapped to environment variables using <code>${{...}}</code> magic.</p>
<p>You could try using <code>toJSON(...)</code> on each of the contexts in a step to see what’s available.</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/actions/learn-github-actions/contexts">
  <header class="source">
      <img src="https://docs.github.com/assets/cb-803/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/actions/learn-github-actions/contexts" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2119212/181099349-4ef02faf-5bb5-46ce-9e83-ffa47b3df86d.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/actions/learn-github-actions/contexts" target="_blank" rel="noopener">Contexts - GitHub Docs</a></h3>

  <p>You can access context information in workflows and actions.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by ViacheslavKudinov** [comment, marked as answer]
<p>Yes, <a class="mention" href="/jsoref">@jsoref</a> is right there.</p>
<p>It’s not exactly env var and not for all GH events, but in this way you could access some context for few events.</p>
<p>If to use events eg <code>issues</code> <a href="https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issues" class="inline-onebox" rel="noopener nofollow ugc">Events that trigger workflows - GitHub Docs</a> or <code>issue_comment</code> <a href="https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment" class="inline-onebox" rel="noopener nofollow ugc">Events that trigger workflows - GitHub Docs</a> some information will be inside payload, for example for <code>issues</code> <a href="https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#issues" class="inline-onebox" rel="noopener nofollow ugc">Webhook events and payloads - GitHub Docs</a>.<br>
Then inside <code>${{ github.event.issue.title }}</code> will be title of issue that triggered event.</p>
<p><a class="mention" href="/kouwakai">@KouWakai</a> you can check this <a href="https://github.community/t/get-contents-body-of-issue-in-action/136057" class="inline-onebox">Get contents/body of issue in action</a> as well</p>

**Comment by KouWakai** [comment]
<p>This one is self resolved.</p>
<p>There is no way to get issue title like I asked.</p>
<p>We could get title from payload which comes from <code>@actions/github</code></p>
<p>like this:</p>
<p>const github = require(’<span class="mention">@actions</span>/github’);</p>
<pre><code class="lang-auto">
    // Get context and payload
    const context = github.context;
    const payload = context.payload;

    if (payload.issue &amp;&amp; payload.issue.title) {
      return payload.issue.title;
    }</code></pre>

**Comment by KouWakai** [comment]
<p><a class="mention" href="/viacheslavkudinov">@ViacheslavKudinov</a> <a class="mention" href="/jsoref">@jsoref</a></p>
<p>Thank you for helping me out:pray:</p>
<p>I couldn’t see reply yesterday for some reason(maybe the replies are meant to be awaiting for approval?)</p>
<p>I’ll check <code>toJSON(...)</code> out.</p>
